client/cli/src/params/offchain_worker_params.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+
+//! Offchain worker related configuration parameters.
+//!
+//! A subset of configuration parameters which are relevant to
+//! the inner working of offchain workers. The usage is solely
+//! targeted at handling input parameter parsing providing
+//! a reasonable abstraction.
+
+use structopt::StructOpt;
+use sc_service::config::OffchainWorkerConfig;
+use sc_network::config::Role;
+
+use crate::error;
+use crate::OffchainWorkerEnabled;
+
+
+/// Offchain worker related parameters.
+#[derive(Debug, StructOpt, Clone)]
+pub struct OffchainWorkerParams {
+	/// Should execute offchain workers on every block.
+	///
+	/// By default it's only enabled for nodes that are authoring new blocks.
+	#[structopt(
+		long = "offchain-worker",
+		value_name = "ENABLED",
+		possible_values = &OffchainWorkerEnabled::variants(),
+		case_insensitive = true,
+		default_value = "WhenValidating"
+	)]
+	pub enabled: OffchainWorkerEnabled,
+
+	/// Allow access to offchain workers indexing API
```

Comment:
where does the term indexing here comes from? It doesn't really connect for me to the description in the next line. Indexing implies some sort of caching similar to databases. Yet the description just says: `Enables a runtime to write directly to ..`. 

Maybe something like: `ENABLE_OFFCHAIN_IMPORT_DB`

Reply:
The data that is going to be written is in some form a cache for i.e. slashing based on this historical data. I am not fixed on the terminology here.

Reply:
@tomusdrw I don't have opinion about this, I would prefer to keep it mostly as is to drive this to completion.

Reply:
So this API allows the block import to insert something to offchain DB, so that it's available later.
1. We want to emphasize, this is a write-only API, so I don't think that it should involve names like DB access or anything like that.
2. It allows you to produce some data that forms something like an index of imported blocks. IMHO indexing emphasizes that the data is going to be complete (i.e. we can produce an index of all transactions in blocks accesible externaly via RPC, etc).
3. I'm a bit reluctant to use "offchain" too much, to avoid the confusion with offchain workers. As @drahnr mentioned, these 2 concepts are pretty much disjoint (indexed data is already available via offchain-db RPCs, so you can access it externally). There are use cases however that will use both APIs - in such case whatever is put into the offchain index is consumed by offchain workers.

I'm not super happy with the naming, open to any suggestions. If we keep the current one, I'd like to request more attention to details and keep everything consistent, i.e.:
we should always use a full name Offchain Indexing API (or just Indexing API), but never mix it up with "workers", cause it has nothing to do.
The concepts we have:
1. Off-chain DB - database that is not part of the consensus (i.e not trie)
2. Off-chain Workers - an opt-in piece of code being run after block import, which has access to extended set of APIs (Off-chain (Worker) APIs), that includes access to Off-chain DB.
3. Off-chain indexing API - an opt-in API that allows Off-chain DB to be populated (write) by block import.